requests_ntlm/requests_ntlm.py

```diff
@@ -134,10 +138,17 @@ def retry_using_http_NTLM_auth(
 
         auth_strip = auth_type + " "
 
-        ntlm_header_value = next(
+        challenges = [
             s
             for s in (val.lstrip() for val in auth_header_value.split(","))
             if s.startswith(auth_strip)
+        ]
+
+        if not challenges:
+            raise AccessDenied("Access denied")
+
+        ntlm_header_value = next(
+            challenges
         ).strip()
```

Comment:
I would use a default for `next` that is `None` here and make the exception a bit more descriptive. Finally I would probably just use the builtin error `PermissionError` rather than defining a new error type here.

```suggestion
        ntlm_header_value = next(
            (
                s
                for s in (val.lstrip() for val in auth_header_value.split(","))
                if s.startswith(auth_strip)
            ),
            None
        ).strip()
        
        if not ntlm_header_value:
            raise PermissionError("Access denied: Server did not respond with NTLM challenge token")
```